js/setupModal.js

```diff
@@ -2,55 +2,79 @@ import './miniatures.js';
+const socialComments = bigPicture.querySelector('.social__comments');
+const commentsCount = bigPicture.querySelector('.comments-count');
+const body = document.body;
 
-// Открытие модального окна
-const showModal = (url, likes, comments, description) => {
-  bigPicture.classList.remove('hidden');
-  const commentsContainer = document.createDocumentFragment();
-  const socialComments = bigPicture.querySelector('.social__comments');
-  socialComments.innerHTML = '';
+const setupComments = (comments) => {
+  let count = 5;
+  const visibleComments = [];
 
-  // После открытия окна тегу <body> добавляется класс modal-open, чтобы контейнер с фотографиями позади не прокручивался при скролле.
-  const body = document.body;
-  body.classList.add('modal-open');
-
-  // После открытия окна прячем блоки счётчика комментариев и загрузки новых комментариев
-  const socialCommentCount = bigPicture.querySelector('.social__comment-count');
-  socialCommentCount.classList.add('hidden');
-  const commentsLoader = bigPicture.querySelector('.comments-loader');
-  commentsLoader.classList.add('hidden');
-
-  comments.forEach(({avatar, message, name}) => {
-    const comment = templateComment.cloneNode(true);
-    comment.querySelector('.social__picture').src = avatar;
-    comment.querySelector('.social__text').textContent = message;
-    comment.querySelector('.social__picture').alt = name;
-    commentsContainer.appendChild(comment);
-  });
+  const checkCommentsQuantity = () => comments.length > 5;
 
-  socialComments.appendChild(commentsContainer);
+  const renderComments = (commentsArr) => {
+    const commentsContainer = document.createDocumentFragment();
 
-  bigPicture.querySelector('.big-picture__img img').src = url;
-  bigPicture.querySelector('.social__likes').textContent = likes;
-  bigPicture.querySelector('.comments-count').textContent = comments;
-  bigPicture.querySelector('.social__caption').textContent = description;
+    commentsArr.forEach(({avatar, message, name}) => {
+      const comment = templateComment.cloneNode(true);
+      comment.querySelector('.social__picture').src = avatar;
+      comment.querySelector('.social__text').textContent = message;
+      comment.querySelector('.social__picture').alt = name;
+      commentsContainer.appendChild(comment);
+    });
+    socialComments.appendChild(commentsContainer);
+  };
+  if (checkCommentsQuantity(comments)) {
+    loadMoreButton.classList.remove('hidden');
+    const commentsToShow = comments.slice(0, 5);
+    visibleComments.push(...commentsToShow);
+    renderComments(commentsToShow);
+    loadMoreButton.addEventListener('click', () => {
+      const commentsArr = comments.slice(count, count + 5);
+      count += 5;
+      renderComments(commentsArr);
+      visibleComments.push(...commentsArr);
+      if (comments.length === visibleComments.length) {
+        loadMoreButton.classList.add('hidden');
+      }
+    });
```

Comment:
```suggestion
    }, { once: true });
```
При открытии и закрытии разных фотографий обработчики всё равно копятся на этой кнопке.
Можно сделать этот обработчик одноразовым, и в обработчике вызывать `setupComments` с оставшимися комментариями.